packages/nuxt/src/components/scan.ts

```diff
@@ -95,7 +95,11 @@ export async function scanComponents (dirs: ComponentsDir[], srcDir: string): Pr
       const suffix = (mode !== 'all' ? `-${mode}` : '')
       const componentNameSegments = resolveComponentNameSegments(fileName.replace(/["']/g, ''), prefixParts)
       const pascalName = pascalCase(componentNameSegments)
-
+      
+      if (pascalName.startsWith('Lazy')) {
```

Comment:
we should use a regexp that ensures it is a complete chunk. e.g. `/^Lazy(?=[A-Z])/`

Reply:
We can do that of course, but I can't seem to think of any examples of components that won't get correctly captured without the additional positive lookahead. Do you think it's necessary?

Reply:
I gave some examples here: https://github.com/nuxt/nuxt/pull/27838#issuecomment-2191302610.

Reply:
Yes, but would they not capture correctly with the beginning check? It already checks that it starts with Lazy regardless of what it has afterwards, as that's also the pattern in loader.ts

Reply:
I realise it's not very likely, but we should only warn if the first _segment_ is lazy, so `lazyn-component.vue` should not trigger this warning.

Reply:
I've looked at the loader pattern again, it just looks for Lazy and then more characters, so a component named Lazyn-component.vue would still capture the Lazy part and read it as n-component.vue, wouldn't it? That means it would still be modified because it has that name conflict, unless I'm missing something?

Reply:
It would capture it with the regexp but the check on the next line would fail as no component is found. So it is safe.

Reply:
I see, so what is the difference in scan.ts? If it begins with Lazy it means it will be modified in loader.ts which would resolve to no component. 

And sorry for all the questions, I'm just trying to understand, I don't mean for it to sound dismissive or anything

Reply:
Just checked:
https://stackblitz.com/edit/github-5spadh?file=components%2FLazyn.vue,app.vue

Even if a component only begins with Lazy it fails to resolve, meaning that chunk regex would actually not capture it and won't display the warning
![image](https://github.com/nuxt/nuxt/assets/92037085/babaf070-e0f8-44f4-9b75-8b4423336424)

Reply:
Good catch! We should update the regexp in the loader too then

Reply:
I mean, as long as the component doesn't begin with Lazy (at all, no matter how many chunks it consists of), it would not cause any issues, that's why it only errors when the component contains Lazy. The question is, does the regex need to be updated, or is it more so about enforcing that components will not begin with "lazy" which can be documented through the warning? Because by changing the loader regex, it's indirectly allowing components to contain lazy without it breaking the app, like I did in my other PR about this. There's also edge cases like LazyN.vue which would conform to the pattern but still break because it still has the Lazy part

Reply:
Yes, it's important to update the regexp; it fixes the bug you found with loading components like `LazynessThing`.

Reply:
@danielroe I just noticed a possible regression with the regex:
What if the component is used like [a-z]*- as opposed to [A-Z][a-z]*?
(For example, how you can use lazy-something and LazySomething)